### PR TITLE
Make component customizable

### PR DESCRIPTION
**Pull request by Fafruch**
TODO: Add API description.

It covers #6 but not #7.

### CODE REVIEW

File: src/lib/utils/computeRRule/computeRRule.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { assign } from 'lodash';
+
+import computeRepeat from './computeRepeat';
+import computeEnd from './computeEnd';
+
+const computeRRule = ({ repeat, end }) => {
+  const repeatObject = computeRepeat(repeat);
+  const endObject = computeEnd(end);
+
+  const rruleObject = assign({}, repeatObject, endObject);
```
Comment (tb) [inline src/lib/utils/computeRRule/computeRRule.js]: You seem to build new objects with `assing({},...)` in many palces in system... consider using deconstriction like `const rruleObject = { ...repeatObject, ...endObject };`

File: src/lib/utils/configureInitialState.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  end: ['On date'],
+};
+
+const configureState = (config = {}) => {
+  const configureFrequency = () => {
+    if (config.repeat) {
+      return config.repeat[0];
+    }
+    return 'Yearly';
+  };
```
Comment (tb) [inline src/lib/utils/configureInitialState.js]: Consider writing less code i.e. `const configureFrequency = () => config.repeat ? config.repeat[0] : 'Yearly'`

File: src/lib/components/Repeat/Yearly/On.js [hunk 3]
```diff
@@ -1,64 +1,78 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import moment from 'moment';
+
 import numericalFieldHandler from '../../../utils/numericalFieldHandler';
 
 const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
```
Comment (pokiujf) [inline src/lib/components/Repeat/Yearly/On.js]: BTW The same array is in `src/lib/utils/computeRRule/computeYearlyOn.js`.
What about extracting this and other hardcoded data you know about, into `lib/constants` ?

File: src/lib/utils/computeRRule/computeRRule.js (hunk 1 above)
Comment (Fafruch) [reply src/lib/utils/computeRRule/computeRRule.js]: Sure, I don't know why I didn't use destructuring. Probably I wasn't sure how to write this:
```js
assign(
    {},
    { freq: RRule.MONTHLY },
    { interval },
    mode === 'on' ? computeMonthlyOn(on) : computeMonthlyOnThe(onThe),
  );
```
with spread syntax.

So for now I removed all lodash'es assign(), thanks for pointing that!

File: src/lib/utils/configureInitialState.js (hunk 2 above)
Comment (Fafruch) [reply src/lib/utils/configureInitialState.js]: It was initially in this form as I didn't know if these functions would extend.
Zipped

File: src/lib/components/End/OnDate.js [hunk 4]
```diff
@@ -3,33 +3,45 @@ import PropTypes from 'prop-types';
...
+        locale={locale}
+      />
+    </div>
+  );
+};
 
 EndOnDate.propTypes = {
   onDate: PropTypes.string.isRequired,
+  config: PropTypes.shape({
+    calendarFirstDayOfAWeek: PropTypes.oneOf(['Mon', 'Sun']),
```
[the remark this answers is not in the document]
Comment (Fafruch) [reply src/lib/components/End/OnDate.js]: Fixed. ;)

File: src/lib/components/End/index.js [hunk 5]
```diff
@@ -10,39 +10,45 @@ const End = ({
...
+            </strong>
+          </label>
+        </div>
+        <div className="col-sm-3">
+          <select
+            name="end.mode"
+            id="End"
+            className="form-control"
+            value={mode}
+            onChange={event => handleChange(event)}
```
[the remark this answers is not in the document]
Comment (Fafruch) [reply src/lib/components/End/index.js]: Yup. ;) Removed that

File: src/lib/components/End/index.js [hunk 6]
```diff
@@ -51,6 +57,10 @@ End.propTypes = {
     onDate: PropTypes.string.isRequired,
   }).isRequired,
   handleChange: PropTypes.func.isRequired,
+  config: PropTypes.shape({
+    end: PropTypes.arrayOf(PropTypes.oneOf(['Never', 'After', 'On date'])),
```
[the remark this answers is not in the document]
Comment (Fafruch) [reply src/lib/components/End/index.js]: I'll consider that yet

File: src/lib/components/ReactRRuleGenerator.js [hunk 7]
```diff
@@ -1,107 +1,98 @@
...
-        after: 1,
-        onDate: moment().format(DATE_TIME_FORMAT),
-      },
-    },
-    isCopied: false,
-  };
+  state = configureInitialState(this.props.config);
+
+  componentWillMount() {
+    this.setState({ ...this.state, rrule: computeRRule(this.state.data) });
```
[the remark this answers is not in the document]
Comment (Fafruch) [reply src/lib/components/ReactRRuleGenerator.js]: You're right! Moved to `configureInitialState`

File: src/lib/components/ReactRRuleGenerator.js [hunk 8]
```diff
@@ -1,107 +1,98 @@
...
+
+    this.setState({ isCopied: true });
+
+    onRRuleCopy(this.state.rrule);
+  }
 
   render() {
+    const { handleChange, handleCopy } = this;
+    const { data, isCopied, rrule } = this.state;
+    const { repeat, end } = data;
```
[the remark this answers is not in the document]
Comment (Fafruch) [reply src/lib/components/ReactRRuleGenerator.js]: Yeah, right : )

File: src/lib/components/Repeat/Monthly/index.js [hunk 9]
```diff
@@ -12,31 +12,52 @@ const RepeatMonthly = ({
...
-          aria-label="Repeat monthly interval"
-          className="form-control"
-          value={interval}
-          onChange={numericalFieldHandler(handleChange)}
+
+      {isOptionAvailable('on') && (
+        <RepeatMonthlyOn
+          mode={mode}
+          on={on}
+          isNotTheOnlyOneMode={!isTheOnlyOneMode('on')}
```
[the remark this answers is not in the document]
Comment (Fafruch) [reply src/lib/components/Repeat/Monthly/index.js]: Changed to `hasMoreModes`

File: src/lib/components/Repeat/Yearly/On.js (hunk 3 above)
Comment (Fafruch) [reply src/lib/components/Repeat/Yearly/On.js]: I thought this day will finally arrive to have a purpose moving it to constants. This day arrived already. Moved to constants. ;)

File: src/lib/constants/index.js [hunk 10]
```diff
@@ -1 +1,5 @@
 export const DATE_TIME_FORMAT = 'YYYY-MM-DD';
+
+export const MONTHS = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+export const DAYS = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday',
```
Comment (pokiujf) [inline src/lib/constants/index.js]: One step further. Does moment.js provide such constants already?

**Pull request merged**